src/new-components/checkbox-field/checkbox-field.stories.mdx

```diff
@@ -54,15 +55,52 @@ export const Template = ({
     disabled,
     indeterminate,
     label,
+    icon,
     'aria-controls': ariaControls,
     'aria-describedby': ariaDescribedby,
     'aria-label': ariaLabel,
     'aria-labelledby': ariaLabelledby,
-}) => <CheckboxField disabled={disabled} indeterminate={indeterminate} label={label} />
+}) => <CheckboxField disabled={disabled} indeterminate={indeterminate} label={label} icon={icon} />
+
+### Checkbox with icon
+
+A checkbox field with an icon and a string. The icon is rendered to the left of the label.
+
+#### Image as Icon
+
+<Canvas>
+    <Story
+        parameters={{ docs: { source: { type: 'dynamic' } } }}
+        name="Image as icon"
+        args={{
+            label: 'Label with icon',
+            icon: '😄',
+        }}
+    >
+        {Template.bind({})}
+    </Story>
+</Canvas>
+
+#### ReactChild as Icon
+
+In this example we are passing an SVG image as icon.
+
+<Canvas>
+    <Story
+        parameters={{ docs: { source: { type: 'dynamic' } } }}
+        name="ReactChild as icon"
+        args={{
+            label: 'Label with icon',
+            icon: <Icon />,
+        }}
+    >
+        {Template.bind({})}
+    </Story>
+</Canvas>
 
 ### ReactNode as Label
```

Comment:
I would suggest that you can remove this section entirely. No need to show in storybook how to do something the wrong way, now that we have a better way to do it.

Reply:
While we don't want to encourages devs to sneak in an `icon` in the label field, I was thinking there might still be some cases when it might be a valid use case to pass a "ReactNode as Label":

- If that's the case (there are some valid use cases), maybe I should change the example to not use `<Icon>` but another valid use case (which I cannot think of right now 😅).

- If that's not the case (no valid use cases to pass in "ReactNode as Label"), then maybe we should change the type of `label` to just text so that devs cannot pass a `ReactNode` anymore.

Reply:
Makes sense.

A more valid use case could be showing how you can use `<strong>` or `<em>` in a label:

```jsx
<CheckboxField
  label={<>Accept our <strong>terms of service</strong></>}
/>
```